### ISSUE TITLE
Session data missing from websocket requests in IE

### ISSUE THREAD

**Issue by rjclardy** [pakyow/pakyow#216]
Something is causing session data to be missing from websocket requests made in IE11.  The session hashes below are the result of performing the same websocket request in Chrome vs IE11...

Chrome:
```
{"session_id"=>"3a9f3e2c2d65a5c257e6f6dcc7710a9ff452f338636d15d44669d44430e5186f", "csrf"=>"fXvzqu1V6KkjZUvlllz2G1Q2lAFLG2VZUl4p0NwuQNk=", "tracking"=>{"HTTP_USER_AGENT"=>"34eb6984ac6d41857a5d530d847e968c63ee6377", "HTTP_ACCEPT_LANGUAGE"=>"66eae971492938c2dcc2fb1ddc8d7ec3196037da"}, "user"=>5, "program"=>1, "socket_key"=>"17602a6401409d0625a48617bd82952bba88d7abcd97b3947f01a722cd8af1fe"}
```

IE11:
```
{"socket_key"=>"7ab0acd36d1dcdc866c984ca4270ad65d5024e78366bf06792cdc170b6024de2"}
```
FWIW I confirmed that the session hash of regular HTTP requests in IE11 looks the same as it does in Chrome - so it does appear to be limited to websockets.

**Comment by bryanp** [comment]
I think `socket_key` is probably being set during the websocket connection request, so it looks like the entire session is unavailable using IE (at least for requests to establish a websocket). It might be that IE doesn't send the cookies as part of this request, which means no session data. Since we get at the session from Rack, seems like we should start there.

**Comment by rjclardy** [comment]
And before I forget, during our pair session we wanted to see the result of [this line](https://github.com/rack/rack/blob/1.6.5/lib/rack/session/cookie.rb#L144) in the rack lib. I was able to confirm that it does return the full session hash in IE when processing an HTTP request but we don't end up hitting that line when making the websocket request.

**Comment by bryanp** [comment]
Ah, that could (in theory) mean the session cookie doesn't exist so it doesn't try and load the data. Just a guess, I haven't verified that this is the case.

**Comment by bryanp** [comment]
Turns out this isn't a problem in Pakyow, but we learned some things that I wanted to share with everyone. If you're using `Rack::Protection` in your app, the session won't be available in WebSockets on IE or Safari. The reason is that neither one sends the `Accepts-Language` header, which `Rack::Protection:: SessionHijacking` uses by default in its validation.

To workaround this, simply change what headers are validated:

```ruby
builder.use Rack::Protection, track: %w[HTTP_USER_AGENT]
```

This makes it less secure, but to my knowledge there's no other workaround at this point.
